src/atoms/fileActions.test.ts

```diff
@@ -0,0 +1,329 @@
+
+    act(() => openFile.current(fileToOpen));
+
+    expect(leftPane.current.id).toBe('LEFT' as PaneId);
+    expect(leftPane.current.files.length).toBe(1);
+    expect(leftPane.current.files).toContainEqual(fileToOpen);
+    expect(leftPane.current.activeFile).toEqual(fileToOpen);
+  });
+
+  test('should open PDF file in the right pane, and be active', () => {
+    const store = createStore();
+    const { result: openFile } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: rightPane } = renderHook(() => useAtomValue(rightPaneAtom, { store }));
+
+    const fileToOpen = makeFile('file.pdf');
+    expect(rightPane.current.files).not.toContainEqual(fileToOpen);
+    expect(rightPane.current.activeFile).toBeUndefined();
+
+    act(() => openFile.current(fileToOpen));
+
+    expect(rightPane.current.id).toBe('RIGHT' as PaneId);
+    expect(rightPane.current.files.length).toBe(1);
+    expect(rightPane.current.files).toContainEqual(fileToOpen);
+    expect(rightPane.current.activeFile).toEqual(fileToOpen);
+  });
+
+  test('should open file once in same panel', () => {
+    const store = createStore();
+    const { result: openFile } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: leftPane } = renderHook(() => useAtomValue(leftPaneAtom, { store }));
+
+    const fileToOpen = makeFile('file.txt');
+    expect(leftPane.current.files.length).toBe(0);
+
+    act(() => {
+      openFile.current(fileToOpen);
+      openFile.current(fileToOpen);
+      openFile.current(fileToOpen);
+    });
+
+    expect(leftPane.current.files.length).toBe(1);
+  });
+
+  test('should close opened file in same pane', () => {
+    const store = createStore();
+    const { result: openFile } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: closeFileFromPane } = renderHook(() => useSetAtom(closeFileFromPaneAtom, { store }));
+    const { result: leftPane } = renderHook(() => useAtomValue(leftPaneAtom, { store }));
+
+    const fileA = makeFile('fileA.txt');
+    const fileB = makeFile('fileB.txt');
+
+    act(() => {
+      openFile.current(fileA);
+      openFile.current(fileB);
+    });
+    expect(leftPane.current.files.length).toBe(2);
+
+    act(() => {
+      closeFileFromPane.current({ paneId: leftPane.current.id, fileId: fileA.path });
+    });
+
+    expect(leftPane.current.files.length).toBe(1);
+  });
+
+  test('should not fail trying to close file that is not opened in pane', () => {
+    const store = createStore();
+    const { result: openFile } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: closeFileFromPane } = renderHook(() => useSetAtom(closeFileFromPaneAtom, { store }));
+    const { result: leftPane } = renderHook(() => useAtomValue(leftPaneAtom, { store }));
+
+    const fileA = makeFile('fileA.txt');
+    const fileB = makeFile('fileB.txt');
+    const fileC = makeFile('fileC.txt');
```

Comment:
I think there is no need to create 3 files, this only makes the test have to check more things than it should